### PR TITLE
Create `.gitignore` when execute init command

### PR DESCRIPTION
**Pull request by sottar**
PR for #953 

## Summary
Automatically generate a `.gitignore` file when execute init command and select `element` or `application`.

 - [x] CHANGELOG.md has been updated

### CODE REVIEW

File: gulpfile.js [hunk 1]
```diff
@@ -37,7 +37,8 @@ gulp.task('compile', () => {
   let tsReporter = typescript.reporter.defaultReporter();
   return mergeStream(
              tsProject.src().pipe(tsProject(tsReporter)),
-             gulp.src(['src/**/*', '!src/**/*.ts']))
+             gulp.src(['src/**/*', '!src/**/*.ts']),
+             gulp.src(['src/**/.gitignore']))
```
Comment (FredKSchott) [inline gulpfile.js]: This like should be unnecessary, no need to compile .gitignore with typescript

**Comment by sottar** [comment]
@FredKSchott Thanks for your reviews !
I added `copy` task in gulpfile.  It just copy `.gitignore` to `lib` directory.

**Comment by stramel** [comment]
Did yeoman make it to where it doesn't ignore hidden files? I remember looking at this a while back and thought it would ignore hidden files. 

@sottar have you checked this?

**Comment by sottar** [comment]
@stramel 
I confirmed that a `.gitignore` file was generated in my branch.

**Comment by justinfagnani** [comment]
@FredKSchott I restarted the tests and they pass... lgty?

**Comment by stramel** [comment]
Fixes #531 as well

**Pull request merged**